docs/usage/caching.md

```diff
@@ -0,0 +1,245 @@
+# Caching
```

Comment:
The page generally looks good, except for some parts which don't make a lot of sense. It's also a bit too verbose. This would have to be reworked (I will provide more details).